### ISSUE TITLE
Explore repackaging of MLMD python wheel as a pure lib

### ISSUE THREAD

**Issue by tarilabs** [opendatahub-io/model-registry-bf4-kf#228]
**Is your feature request related to a problem? Please describe.**
MLMD python wheel dependency (used by this MR python client) is platform/architecture specific

**Describe the solution you'd like**
MR python client to rely on pure library for supporting the pragmatic use-case leveraged 

**Describe alternatives you've considered**
This is based on the observation that we are using MLMD with gRPC remote connection, so the embedded binary executable in the MLMD python wheel is not strictly required (with caveat of tests, but this is mentioned in details in the document produced by this exploration).

The document under discussion presents now 2 alternatives, which can help Apple-silicon contributors of Model Registry, but also can potentially simplify our life when we will face deployment of the Model Registry python client on non-x86 platforms (I'm thinking of Edge scenarios, etc).

The alternatives ~are:

- Repackaging the distributed wheel, making changes to make it pure lib
- Changing pip+bazel build of upstream MLMD project, by making that produce a pure lib

**Additional context**
Pre-requisites:
- #225

**Comment by rareddy** [comment]
@tarilabs now that we want to use the soft fork, is this something @isinyaaa can take on as a task?

**Comment by tarilabs** [comment]
I had first to get the prerequisite accepted by the group (https://github.com/opendatahub-io/model-registry/pull/225#event-11257168316)
and I've kept tabs over the holidays if any update was moving for (`https://github.com/google/ml-metadata/issues/143#issuecomment-1871614177`)

My recommendation is as follows:
1. I've raised our ADR to document our decision in our repo: https://github.com/opendatahub-io/model-registry/pull/261
2. I will ask for creation of soft-fork repo in ODH, to ODH maintainers
3. I will perform the same steps as done in experimental own-repo but on step2
![image](https://github.com/opendatahub-io/model-registry/assets/1699252/7f384153-c8c3-48db-8d98-7e75fb040c95)
4. I will coordinate with @isinyaaa to wire analogous publishing of PyPI from the repo

wdyt?
